rskj-core/src/test/java/co/rsk/peg/bitcoin/PocSighashTest.java

```diff
@@ -473,6 +473,67 @@ void test_each_input_sighash_is_unique_for_a_signed_erp_tx_testnet() {
         }
     }
 
+    @Test
+    void test_redeemScript_can_be_obtained_from_input() {
+        // Arrange
+        NetworkParameters networkParameters = BridgeTestNetConstants.getInstance().getBtcParams();
+        int erpFedActivationDelay = 720;
+
+        List<FedSigner> fedMembers = FedSigner.listOf("federator1", "federator2", "federator6");
+        List<FedSigner> erpFedMembers = FedSigner.listOf("erp-fed-01", "erp-fed-02", "erp-fed-03");
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        P2shErpFederation fed = new P2shErpFederation(
+            fedMembers.stream().map(FedSigner::getFed).collect(Collectors.toList()),
+            Instant.now(),
+            0L,
+            networkParameters,
+            erpFedMembers.stream().map(FedSigner::getFed).map(FederationMember::getBtcPublicKey).collect(Collectors.toList()),
```

Comment:
```suggestion
            erpFedMembers.stream().map(FedSigner::getFed).map(FederationMember::getBtcPublicKey).collect(Collectors.toList()),
```
Lets Extract This To A List